tests/cmdline/commands/test_process.py

```diff
@@ -376,74 +382,54 @@ def test_multiple_processes(self):
+    await_condition(lambda: node.paused)
 
-    test_daemon_timeout = 5.
-    start_time = time.time()
-    while calc.process_state is not plumpy.ProcessState.WAITING:
-        if time.time() - start_time >= test_daemon_timeout:
-            raise RuntimeError('Timed out waiting for process to enter waiting state')
-
-    # Make sure that calling any command on a non-existing process id will not except but print an error
-    # To simulate a process without a corresponding task, we simply create a node and store it. This node will not
-    # have an associated task at RabbitMQ, but it will be a valid `ProcessNode` with and active state, so it will
-    # pass the initial filtering of the `verdi process` commands
-    orphaned_node = WorkFunctionNode()
-    orphaned_node.set_process_state(ProcessState.RUNNING)
-    orphaned_node.store()
-    non_existing_process_id = str(orphaned_node.pk)
-    for command in [process_pause, process_play, process_kill]:
-        result = run_cli_command(command, [non_existing_process_id])
-        assert 'Error:' in result.output
-
-    assert not calc.paused
-    result = run_cli_command(process_pause, [str(calc.pk)])
-
-    # We need to make sure that the process is picked up by the daemon and put in the Waiting state before we start
-    # running the CLI commands, so we add a broadcast subscriber for the state change, which when hit will set the
-    # future to True. This will be our signal that we can start testing
-    waiting_future = Future()
-    filters = kiwipy.BroadcastFilter(
-        lambda *args, **kwargs: waiting_future.set_result(True), sender=calc.pk, subject='state_changed.*.waiting'
-    )
-    runner.communicator.add_broadcast_subscriber(filters)
-
-    # The process may already have been picked up by the daemon and put in the waiting state, before the subscriber
-    # got the chance to attach itself, making it have missed the broadcast. That's why check if the state is already
-    # waiting, and if not, we run the loop of the runner to start waiting for the broadcast message. To make sure
-    # that we have the latest state of the node as it is in the database, we force refresh it by reloading it.
-    calc = load_node(calc.pk)
-    if calc.process_state != plumpy.ProcessState.WAITING:
-        runner.loop.run_until_complete(asyncio.wait_for(waiting_future, timeout=5.0))
-
-    # Here we now that the process is with the daemon runner and in the waiting state so we can starting running
-    # the `verdi process` commands that we want to test
-    result = run_cli_command(process_pause, ['--wait', str(calc.pk)])
-    assert calc.paused
-
-    if cmd_try_all:
-        cmd_option = '--all'
-    else:
-        cmd_option = str(calc.pk)
-
-    result = run_cli_command(process_play, ['--wait', cmd_option])
-    assert not calc.paused
-
-    result = run_cli_command(process_kill, ['--wait', str(calc.pk)])
-    assert calc.is_terminated
-    assert calc.is_killed
+
+@pytest.mark.requires_rmq
```

Comment:
The only thing speaking against dividing the tests might have been runtime, but I checked and they don't show up in top runtime list https://github.com/aiidateam/aiida-core/runs/8232871620?check_suite_focus=true#step:10:2706

Reply:
I made the fixtures such that they only start the daemon when needed so as not to incur unnecessary calls for subsequent tests needing the same env. And with that guarantee, it is better to have the tests separated compared to a single enormous tests doing multiple asserts. Can't really call it a unit tests anymore then

Reply:
Yep, all good